tests/trainer/test_trainer.py

```diff
@@ -1856,3 +1856,36 @@ def test_check_val_every_n_epoch_exception(tmpdir):
             max_epochs=1,
             check_val_every_n_epoch=1.2,
         )
+
+
+@pytest.mark.parametrize("datamodule", [False, True])
+def test_trainer_predict_verify_config(tmpdir, datamodule):
```

Comment:
shouldn't this go into tests/trainer/test_config_validator?

Reply:
Thanks, @awaelchli! Updated!